addons/binding/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/internal/convention/homeassistant/ComponentBinarySensor.java

```diff
@@ -57,17 +58,21 @@
     protected Config config = new Config();
 
     public ComponentBinarySensor(ThingUID thing, HaID haID, String configJSON,
-            @Nullable ChannelStateUpdateListener channelStateUpdateListener, Gson gson) {
+            @Nullable ChannelStateUpdateListener channelStateUpdateListener, Gson gson,
+            TransformationServiceProvider provider) {
         super(thing, haID, configJSON, gson);
         config = gson.fromJson(configJSON, Config.class);
 
         if (config.force_update) {
             throw new UnsupportedOperationException("Component:Sensor does not support forced updates");
         }
 
-        channels.put(sensorChannelID,
-                new CChannel(this, sensorChannelID, new OnOffValue(config.payload_on, config.payload_off),
-                        config.state_topic, null, config.name, config.unit_of_measurement, channelStateUpdateListener));
+        CChannel sensorChannel = new CChannel(this, sensorChannelID,
```

Comment:
Note to myself: Use a channel builder pattern

Reply:
I thought about this also.
One of my next request would add specialiezed constructors for read only/write only channels.
But I will thing this through again.